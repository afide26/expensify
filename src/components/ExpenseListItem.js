import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expensesActions';

const ExpenseListItem = ({dispatch, id, description,note, amount, createdAt})=>{
  return(
    <div>
      <h3><strong>{description}</strong></h3>
      <p>Amount due: ${amount/100}.00</p>
      <p>Note: <em>{note}</em></p>
      <code>Created at: {createdAt}</code>
      <button onClick={()=>(dispatch(removeExpense({id})))}>
        Remove
      </button>
      <hr/>
    </div>
  )
}


export default connect()(ExpenseListItem);