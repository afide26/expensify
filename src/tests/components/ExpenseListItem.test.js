import React from 'react';
import ExpenseListItem from '../../components/ExpenseListItem';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expense.fixture';

// const expense = {
//   id:1,
//   description:'Rent',
//   note:'monthly',
//   createdAt: 145405012333,
//   amount:'3000'
// }

const expense = expenses[0];

// Long hand
test("should render the ExpenseListItem with data", ()=>{
  const { id, description, note, amount, createdAt } = expense;
  const wrapper = shallow(<ExpenseListItem 
                            id={id} 
                            description={description}
                            amount={amount}
                            note={note}
                            />);

  expect(wrapper).toMatchSnapshot();
});

// Short hand
test("should render the ExpenseListItem with data", ()=>{
  const wrapper = shallow(<ExpenseListItem {...expense}/>);
  expect(wrapper).toMatchSnapshot();
})