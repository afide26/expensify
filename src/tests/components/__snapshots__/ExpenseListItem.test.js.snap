// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpenseListItem with data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      <strong>
        Gum
      </strong>
    </h3>
  </Link>
  <p>
    Amount due: $
    395
  </p>
  <p>
    Note: 
    <em>
      now
    </em>
  </p>
  <code>
    Created at: 
    January 1st, 1970
  </code>
  <hr />
</div>
`;

exports[`should render the ExpenseListItem with data 2`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      <strong>
        Gum
      </strong>
    </h3>
  </Link>
  <p>
    Amount due: $
    395
  </p>
  <p>
    Note: 
    <em>
      now
    </em>
  </p>
  <code>
    Created at: 
    January 1st, 1970
  </code>
  <hr />
</div>
`;
